apps/keyboard/js/views/swipeable_panel_view.js

```diff
@@ -105,6 +105,20 @@ SwipeablePanelView.prototype.gotoSection = function(index) {
   this._updateIndicator();
 };
 
+SwipeablePanelView.prototype.resize = function resize(totalWidth) {
+  var self = this;
+
+  this.options.totalWidth = totalWidth;
+
+  // Reposition sections when resize;
+  this.sections.forEach(function(section, index) {
+    var translateX = (index === self.currentSectionIndex) ? 0 :
+    (self.currentSectionIndex > index) ? -totalWidth : totalWidth;
+
+    section.style.transform = 'translateX(' + translateX + 'px)';
```

Comment:
What if the user has her/his finger swiping **while** rotate the phone? Is it possible to address that too?